### PR TITLE
Show people how to stub service in acceptance test

### PR DESCRIPTION
**Pull request by jeffrey008**
It took me hours to figure it out. 
I tried to follow things here:https://guides.emberjs.com/v2.15.0/testing/testing-components/#toc_stubbing-services, but it didn't work out for acceptance.
Finally I found a solution here: https://guides.emberjs.com/v2.15.0/tutorial/service/#toc_stubbing-services-in-acceptance-tests
The documentation about testing seems scattering around inside the guides, so I hope putting them back together will save people's time.

### CODE REVIEW

**Comment by jeffrey008** [comment]
Somehow travis is failing me. Cannot install the spell checking gem.

File: source/testing/acceptance.md [hunk 1]
```diff
@@ -39,6 +39,37 @@ test('should add new post', function(assert) {
...
+  email: 'test@testing.com',
+  token: '',
+  isLoggedIn() {
+    return true;
+  },
+});
+
+moduleForAcceptance('Acceptance | login', {
+  beforeEach(){
+    this.application.register('service:user-service', userStub);
```
Comment (rwjblue) [inline source/testing/acceptance.md]: Hmm. I don't think this works correctly due to some shenanigans between `Ember.Application` instances and `Ember.ApplicationInstance` instances.  Basically, I do not believe that this mechanism will allow the stub to clobber and "win" over the actual module that would be resolved (`app/services/user-service.js` in this case).

The thing that I generally do is:

```js
this.register = (...args) => this.application.__deprecatedInstance__.register(...args);
this.lookup = (...args) => this.application.__deprecatedInstance__.lookup(...args);
```

Comment (jeffrey008) [reply source/testing/acceptance.md]: This is what the tutorial says. Perhaps we need to update the tutorial to the one you do?

Comment (toddjordan) [reply source/testing/acceptance.md]: the tutorial seems to stub the services, but if that's not the suggested way, we can change it.  Write up an issue and/or pr and we can get it updated.

Comment (jeffrey008) [reply source/testing/acceptance.md]: I think stubbing the service is okay in acceptance test, but just we need to show people how to do it.

**Comment by toddjordan** [comment]
This PR needs to be revamped now that we have a new way of testing.  See https://guides.emberjs.com/v3.1.0/tutorial/service/#toc_stubbing-services-in-application-tests
for reference.  @jeffrey008 Are you willing to take a stab at it (and sorry this pr has lasted this long)

**Comment by jeffrey008** [comment]
Hi @toddjordan , I haven't touch ember since 2.16, so I am not sure how things goes now.

**Comment by toddjordan** [comment]
no problem @jeffrey008 .  If you find a desire to learn, you are welcome to contribute. The best documentation comes from people that are learning :-)

**Comment by sivakumar-kailasam** [comment]
Thanks for taking the time to contribute to this project. We’re in the process of switching from middleman to an ember app to make it easy for contribution.

We’re closing issue since its been open for too long without progress / it needs more discussion which is probably better had via a new PR in the new app. (sorry about that)

If this PR has changes to the content of guides please raise a PR to https://github.com/ember-learn/guides-source. If the change is to the guides viewer (i.e., the app) then please raise a PR in https://github.com/ember-learn/guides-app/. The READMEs and contributing.md guidelines of those projects will familiarize you with the new process. Let us know if you have any questions!

PS: this is a standard msg we're using across such PRs

**Pull request closed without merging**